### PR TITLE
Move the publish endpoint to publications

### PR DESCRIPTION
**Pull request by daviddavis**
ref #4678
https://pulp.plan.io/issues/4678

### CODE REVIEW

File: pulp_file/app/tasks/publishing.py [hunk 1]
```diff
@@ -27,18 +27,26 @@ def publish(publisher_pk, repository_version_pk):
         repository_version_pk (str): Create a publication from this repository version.
 
     """
-    publisher = FilePublisher.objects.get(pk=publisher_pk)
+    if publisher_pk:
```
Comment (bmbouter) [inline pulp_file/app/tasks/publishing.py]: Are we removing Publishers in a separate PR? IIRC, there was a ticket for that?

Comment (daviddavis) [reply pulp_file/app/tasks/publishing.py]: Yes https://pulp.plan.io/issues/4720

Comment (bmbouter) [reply pulp_file/app/tasks/publishing.py]: Thank you.

File: pulp_file/tests/functional/utils.py [hunk 2]
```diff
@@ -91,6 +92,31 @@ def populate_pulp(cfg, url=FILE_FIXTURE_MANIFEST_URL):
     return client.get(FILE_CONTENT_PATH)['results']
 
 
+def create_publication(cfg, repo, version_href=None, publisher=None):
```
Comment (asmacdo) [inline pulp_file/tests/functional/utils.py]: recommend s/create_publication/create_file_publication/

**Pull request merged**